### ISSUE TITLE
Double callback

### ISSUE THREAD

**Issue by stickx** [ethereum/web3.js#5367]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When there is an exception thrown inside the callback function of any web3 call, the callback is called again with the error.
Super dangerous.

### Expected Behavior

Oldschool web3 (v0.x.x) would throw correctly.

### Steps to Reproduce

```
var Web3 = require('web3new')
var web3 = new Web3();

web3.setProvider(new web3.providers.HttpProvider(`yournode`));

web3.eth.getBlockNumber((err,done) => {
    console.log('callback called', err, done);
    throw 'hi'
})


// web3 v1.x.x output:
// callback called null 15383941
// callback called Error: Invalid JSON RPC response: {"size":0,"timeout":0}

// web3 v0.x.x output:
// callback called null 15383926
// test.js:8
//     throw 'hi'
//     ^

```

### Web3.js Version

1.7.5

### Environment

- Operating System: macosx
- Browser: n/a
- Node.js Version: 14.16.1
- NPM Version: 6.14.12

**Comment by postaenzo** [comment]
Hi , 
I'm receiving the same error using Node v16.14.0 and web3 1.8.1.  The error that I'm receiving it randomically. 

Error: Invalid JSON RPC response: {"size":0,"timeout":0}
    at Object.InvalidResponse (C:\Users\Utente\Documents\GitHub\neecosBOT\node_modules\web3-core-helpers\lib\errors.js:43:16)
    at C:\Users\Utente\Documents\GitHub\neecosBOT\node_modules\web3-providers-http\lib\index.js:126:29
    at processTicksAndRejections (node:internal/process/task_queues:96:5)

**Comment by hodler-smart** [comment]
@postaenzo did you solve this problem?
I have same problem.

**Comment by postaenzo** [comment]
no, still on the same. it's randomic and not easy to replicate

**Comment by Muhammad-Altabba** [comment]
Thanks all for reporting the issue,

I added a note (https://github.com/web3/web3.js/commit/555aa0d212e4738ba7a943bbdb34335518486950#r91563366) to be more investigated by the one who will open an MR for this.

**Comment by Spacesai1or** [comment]
Yea it would seem [these lines](https://github.com/web3/web3.js/blob/12a6d6e4ebf085a687e8624ae71777ab52d6c243/packages/web3-providers-http/src/index.js#L141-L145):

```typescript
response.json().then(function(data) {
    callback(null, data);
}).catch(function(error) {
    callback(errors.InvalidResponse(response));
});
```

should be refactored similarly to how it was before #5179:

```typescript
try {
    result = JSON.parse(result);
} catch (e) {
    error = errors.InvalidResponse(request.responseText);
}

_this.connected = true;
callback(error, result);
```

**Comment by jdevcs** [comment]
Fixed in 1.10.2 patch

**Comment by ghost** [comment]
@stickx @postaenzo @gurujustin Fixed version `1.10.2` is just released, could you try once again?
